src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -745,24 +752,54 @@ winrt::fire_and_forget AppHost::_createNewTerminalWindow(Settings::Model::Global
     co_return;
 }
 
+// Method Description:
+// - Helper to initialize our instance of IVirtualDesktopManager. If we already
+//   got one, then this will just return true. Otherwise, we'll try and init a
+//   new instance of one, and store that.
+// - This will return false if we weren't able to initialize one, which I'm not
+//   sure is actually possible.
+// Arguments:
+// - <none>
+// Return Value:
+// - true iff _desktopManager points to a non-null instance of IVirtualDesktopManager
+bool AppHost::_LazyLoadDesktopManager()
+{
+    if (_desktopManager == nullptr)
+    {
+        try
+        {
+            _desktopManager = winrt::create_instance<IVirtualDesktopManager>(__uuidof(VirtualDesktopManager));
+        }
+        CATCH_LOG();
+    }
+
+    return _desktopManager != nullptr;
+}
+
 void AppHost::_HandleSummon(const winrt::Windows::Foundation::IInspectable& /*sender*/,
-                            const winrt::Windows::Foundation::IInspectable& /*args*/)
+                            const Remoting::SummonWindowBehavior& args)
 {
     _window->SummonWindow();
+
+    if (args != nullptr && args.MoveToCurrentDesktop())
+    {
+        if (_LazyLoadDesktopManager())
+        {
+            GUID currentlyActiveDesktop;
+            VirtualDesktopUtils::GetCurrentVirtualDesktopId(&currentlyActiveDesktop);

```

Comment:
if this fails, GUID is filled with garbage memory. initialize it?

Reply:
can GetCurrentVirtualDesktopId return an HRESULT?

Reply:
okay yea, in debug it's returning `{0}` but I'll do that manually. `GetCurrentVirtualDesktopId` already returns a `bool` so yea I should check that.